### PR TITLE
Add ability to remove server and user

### PR DESCRIPTION
**Pull request by sferhah**
1. Ability to delete a server from the server list,
2. Ability to delete a user from the user list,
3. Hide password chars

### CODE REVIEW

File: src/Jellyfin.Maui/Resources/Styles/DataTemplates.xaml [hunk 1]
```diff
@@ -15,10 +15,32 @@
...
+                    <RowDefinition Height="Auto" />
+                </Grid.RowDefinitions>
+                <Grid.ColumnDefinitions>
+                    <ColumnDefinition Width="*" />
+                    <ColumnDefinition Width="Auto" />
+                </Grid.ColumnDefinitions>
+                <Label Grid.Row="0"
+                       Grid.Column="0"
+                       Text="{Binding Name, Mode=OneTime}" />
+                <Label Grid.Row="0"
```
Comment (crobibero) [inline src/Jellyfin.Maui/Resources/Styles/DataTemplates.xaml]: I would prefer to use a button to delete the server

File: src/Jellyfin.Mvvm/ViewModels/Login/ServerSelectViewModel.cs [hunk 2]
```diff
@@ -67,4 +67,14 @@ private void SelectServer(ServerStateModel? server)
...
         }
     }
+
+    [RelayCommand]
+    private async Task DeleteServerAsync(ServerStateModel? server)
+    {
+        if (server is not null)
+        {
+            await _stateStorageService.RemoveServerAsync(server.Id);
+            InitializeAsync().SafeFireAndForget();
```
Comment (crobibero) [inline src/Jellyfin.Mvvm/ViewModels/Login/ServerSelectViewModel.cs]: Should just `await` this task

File: src/Jellyfin.Mvvm/ViewModels/Login/SelectUserViewModel.cs [hunk 3]
```diff
@@ -73,4 +73,14 @@ private void SelectUser(UserStateModel? user)
...
         }
     }
+
+    [RelayCommand]
+    private async Task DeleteUserAsync(UserStateModel? user)
+    {
+        if (user is not null)
+        {
+            await _stateStorageService.RemoveUserAsync(user.Id, user.ServerId);
+            InitializeAsync().SafeFireAndForget();
```
Comment (crobibero) [inline src/Jellyfin.Mvvm/ViewModels/Login/SelectUserViewModel.cs]: Should just `await` this task

File: src/Jellyfin.Maui/Resources/Styles/DataTemplates.xaml (hunk 1 above)
Comment (sferhah) [reply src/Jellyfin.Maui/Resources/Styles/DataTemplates.xaml]: You sure? I tried first with a button, and it really looks ugly, especially on Android. (cf image: server + button, user + label)

github

Comment (sferhah) [reply src/Jellyfin.Maui/Resources/Styles/DataTemplates.xaml]: With a transparent background, it's now OK on Android, but on windows it has a blue background.
github

Comment (crobibero) [reply src/Jellyfin.Maui/Resources/Styles/DataTemplates.xaml]: I would rather use the native controls when possible instead lf using recognizers on a label.

Just as an fyi- the design is 95% placeholder to just get the functionality working

Comment (sferhah) [reply src/Jellyfin.Maui/Resources/Styles/DataTemplates.xaml]: I suspect you did not see my last commit (The label has been replaced with a button)

**Comment by crobibero** [comment]
`./Jellyfin.Maui/Resources/Styles/DataTemplates.xaml` is failing the xaml lint check

File: src/Jellyfin.Maui/Pages/Login/LoginPage.xaml.cs [hunk 4]
```diff
@@ -16,5 +16,8 @@ public LoginPage(LoginViewModel viewModel)
         : base(viewModel)
     {
         InitializeComponent();
+
+        UserNameEntry.Completed += (s, e) => PasswordEntry.Focus();
+        PasswordEntry.Completed += (s, e) => viewModel.LoginCommand.Execute(null);
```
Comment (crobibero) [inline src/Jellyfin.Maui/Pages/Login/LoginPage.xaml.cs]:
```suggestion
        UserNameEntry.Completed += (_, _) => PasswordEntry.Focus();
        PasswordEntry.Completed += (_, _) => viewModel.LoginCommand.Execute(null);
```

**Review by crobibero — APPROVED** [review]

**Pull request merged**